analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpCognitiveComplexityMetric.cs

```diff
@@ -57,6 +57,29 @@ public override void Visit(SyntaxNode node)
                     State.IncreaseComplexityByNestingPlusOne(switchExpression.SwitchKeyword);
                     State.VisitWithNesting(node, base.Visit);
                 }
+                else if (BinaryPatternSyntaxWrapper.IsInstance(node))
+                {
+                    var nodeKind = node.Kind();
+                    var binaryPatternNode = (BinaryPatternSyntaxWrapper)node;
+                    if (!State.LogicalOperationsToIgnore.Contains(binaryPatternNode)
+                        && (nodeKind == SyntaxKindEx.AndPattern
+                            || nodeKind == SyntaxKindEx.OrPattern))
```

Comment:
I think we can do the node kind checks first since the comparison for them is faster.